### ISSUE TITLE
Question about the safety of the "entries" method

### ISSUE THREAD

**Issue by fabiospampinato** [jakearchibald/idb-keyval#169]
I see the `entries` method basically requests both keys and values and then merges them together:

```ts
return Promise.all([
    promisifyRequest(store.getAllKeys()),
    promisifyRequest(store.getAll()),
]).then(([keys, values]) => keys.map((key, i) => [key, values[i]]));
```

This looks potentially problematic to me, but I don't know enough about IndexedDB to be sure, hence the question: can this ever return an incorrect result? Like is it possible that after the key for an object is returned, but before the value for that object is returned, the key gets deleted so that one ends up with the array of keys and the array of values having different lengths, which would potentially completely mess up the database depending on how those entires are manipulated?

**Comment by jakearchibald** [comment]
No, they're requested as part of the same IDB transaction. A write transaction cannot happen in the middle of a read transaction.
